GramAddict/core/views.py

```diff
@@ -660,6 +660,29 @@ def _getActionBarTitleBtn(self):
             resourceIdMatches=re_case_insensitive, className="android.widget.TextView"
         )
 
+    def getFollowButton(self):
+        button_regex = "android.widget.Button|android.widget.TextView"
+        following_regex = "^Following|^Requested"
+        followback_regex = "^Follow Back$"
+
+        following_button = self.device.find(
+            classNameMatches=button_regex,
+            clickable=True,
+            textMatches=following_regex,
+        )
+        followback_button = self.device.find(
+            classNameMatches=button_regex,
+            clickable=True,
+            textMatches=followback_regex,
+        )
+        if following_button.exists():
+            return following_button, "Following"
+
+        if followback_button.exists():
+            return followback_button, "Follow Back"
+
+        return None, None
```

Comment:
I'd propose a change here. We could split this in two methods, for better separation of concern. On method returns only the uiautomator button, e.g. `getFollowButton()` and another method returns an enum with the button type, e.g. `getFollowButtonType()` -> ButtonFollowEnum { Following, FollowBack, Requested }. This way we can reuse the code better. Method `getFollowButtonType()` may use `getFollowButton` to identify which futton is it.

Reply:
👍